src/v/datalake/record_multiplexer.cc

```diff
@@ -241,6 +241,10 @@ record_multiplexer::operator()(model::record_batch batch) {
 ss::future<result<record_multiplexer::write_result, writer_error>>
 record_multiplexer::end_of_stream() {
     if (_error) {
+        // TODO: Call finish on all writeres to clean up resources. Redpanda
```

Comment:
The crash is only in debug actually.

Reply:
Is this related to the change in this PR?